src/main/java/budgetbuddy/storage/scripts/JsonSerializableScriptDescriptionMap.java

```diff
@@ -0,0 +1,56 @@
+package budgetbuddy.storage.scripts;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import budgetbuddy.model.ScriptLibrary;
+import budgetbuddy.model.attributes.Description;
+import budgetbuddy.model.script.Script;
+import budgetbuddy.model.script.ScriptName;
+
+/**
+ * An immutable script description map that is serialisable to JSON.
+ */
+@JsonRootName("scriptDescriptionMap")
+public class JsonSerializableScriptDescriptionMap {
+    private final Map<String, String> descriptionMap;
+
+    /**
+     * Constructs a {@link JsonSerializableScriptDescriptionMap} from the given map.
+     *
+     * @param descriptionMap the map
+     */
+    public JsonSerializableScriptDescriptionMap(
+            @JsonProperty("descriptionMap") Map<String, String> descriptionMap) {
+        this.descriptionMap = descriptionMap;
+    }
+
+    /**
+     * Constructs a {@link JsonSerializableScriptDescriptionMap} from the scripts in the given library.
+     *
+     * @param library the library
+     */
+    public JsonSerializableScriptDescriptionMap(ScriptLibrary library) {
+        this.descriptionMap = new HashMap<String, String>();
+        for (Script script : library.getScriptList()) {
+            descriptionMap.put(script.getName().toString(), script.getDescription().getDescription());
+        }
+    }
+
+    /**
+     * Retrieves the description map.
+     *
+     * @return the map
```

Comment:
You can leave out obvious parameters/return values if you want, it'll pass the checkstyle tests.

Reply:
Oh, cool. I just fill them in out of habit, cause IDEA pre-creates the `@param`/`@return` automatically when you type `/**` \*shrugs*